src/main/java/io/jenkins/plugins/appcenter/AppCenterRecorder.java

```diff
@@ -131,6 +139,96 @@ private boolean uploadToAppCenter(Run<?, ?> run, FilePath filePath, TaskListener
         }
     }
 
+    private String fetchReleaseNotes(Run<?, ?> build, FilePath filePath, EnvVars vars) {
```

Comment:
I don't want `AppCenterRecorder` to do a lot of logic. It is much easier to test if it simply configures a job and starts the job. The bulk of the logic should be done away from this class. At the moment that would be in `UploadTask` but in the future I am working on breaking that down in to independent tasks. Either way this logic needs to be moved away from this class.